extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/MongoSourceBuilder.java

```diff
@@ -254,15 +256,26 @@ private Batch(
         }
 
         /**
-         * If {@code true}, the lack of database or collection will cause an error.
-         * If {@code false}, database and collection will be automatically created.
-         * Default value is {@code true}.
+         * If non {@link ResourceChecks#NEVER}, the lack of database or collection will cause an error.
+         * Otherwise, database and collection will be automatically created.
+         * Default value is {@link ResourceChecks#ONCE_PER_JOB}.
          *
-         * @param throwOnNonExisting if exception should be thrown when database or collection does not exist.
+         * @param checkResourceExistence if exception should be thrown when database or collection does not exist.
          */
         @Nonnull
-        public Batch<T> throwOnNonExisting(boolean throwOnNonExisting) {
-            params.setThrowOnNonExisting(throwOnNonExisting);
+        public Batch<T> checkResourceExistence(ResourceChecks checkResourceExistence) {
```

Comment:
you kept `throwOnNonExisting` in sink, but not is source. If that is for backward compatibility with docs (https://docs.hazelcast.com/hazelcast/5.3/integrate/mongodb-connector#mongodb-sink) then maybe it would be worthwhile to deprecate `throwOnNonExisting`

Reply:
if you meant that you removed `throwOnNonExisting` from sink builder, then it is still there (and broken).

Reply:
Right, leftover. I removed it.